### PR TITLE
Update eslint

### PR DESCRIPTION
**Pull request by sssmi**
# Description

- Update eslint and its plugins
- Added `eslint-plugin-tailwindcss` plugin
- Tighten up Vue rules from `plugin:vue/vue3-essential` to `plugin:vue/vue3-recommended`
- Use plain eslint plugins instead of vue-cli ones. Running linter on webpack only slows down local development. Running eslint before commit and in CI should suffice.
- Integrate prettier as eslint plugin, so prettier can be used to format Vue components.
- Prettier is now only used to format only .md, .json, .yaml and .html files. Eslint (with prettier integrated) handles .js, .ts and .vue files
- Upgrated husky, and added `lint-staged` to pre-commit hook, so linters are run against staged files.

## Type of change

- [x] Dependency changes
- [x] Code refactor / cleanup

## How should this be tested?

Please provide instructions so we can test. Please also list any relevant details for your test configuration.

Run `npm run lint:fix`. See if the autofixed issues make sense. Also look at the issues that require manual fixing. Do they make sense?

## Visual context

Please provide any relevant visual context for UI changes or additions. This could be static screenshots or a loom screencast.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have requested at least 2 reviews (If the PR is significant enough, use best judgement here)
- [x] I have commented my code where relevant, particularly in hard-to-understand areas
- [x] If package-lock.json has changes, it was intentional.
- [x] The base of this PR is `master` if hotfix, `develop` if not

### CODE REVIEW

File: .husky/pre-commit
Comment (sssmi) [inline .husky/pre-commit:4]: I will uncomment this before merge

File: .eslintrc.js
Comment (sssmi) [inline .eslintrc.js:8]: All these globals are not needed anymore

File: .eslintrc.js
Comment (sssmi) [inline .eslintrc.js:37]: Removed some rules, that don't do anything anymore

File: .eslintrc.js
Comment (sssmi) [inline .eslintrc.js:31]: Vue recommends multi word component names, so they don't get mixed with regular html elements, but many components are already single word, so had to turn this off.

**Review by garethfuller — COMMENTED** [review]
Thanks, looks good. Will you run it now? I can approve after that.

**Review by timjrobinson — APPROVED** [review]
LGTM. What's the reasoning behind passing component properties as dash case rather than camelCase? I thought camelCase made more sense becuase that's how they're defined as properties in the Props object so it's easier to find where they're used, but I'm not too fussed either way.

**Comment by sssmi** [comment]
> LGTM. What's the reasoning behind passing component properties as dash case rather than camelCase? I thought camelCase made more sense becuase that's how they're defined as properties in the Props object so it's easier to find where they're used, but I'm not too fussed either way.

Good point. kebab-case props is the default rule in recommended settings in [vue/attribute-hyphenation](https://eslint.vuejs.org/rules/attribute-hyphenation.html)

Setting the `vue/attribute-hyphenation` to `never` might make more sense.

**Review by garethfuller — APPROVED** [review]
Looks good, I went through some basic testing in the previews and nothing seems to have been affected, nice work! 🙌

**Review by timjrobinson — APPROVED** [review]

**Comment by insulineru** [comment]
Good job, @sssmi 👍🏻

**Pull request merged**
